crates/era/tests/mod.rs

```diff
@@ -0,0 +1,345 @@
+//! Integration tests
+//!
+//! These tests use the `reth-era-downloader` client to download `.era1` files
+//! temporarily and verify that we can correctly read and decompress their data.
+
+use alloy_primitives::{B256, U256};
+use reqwest::{Client, Url};
+use reth_era::{
+    e2s_types::E2sError,
+    era1_file::{Era1File, Era1Reader, Era1Writer},
+    execution_types::MAX_BLOCKS_PER_ERA1,
+};
+use reth_era_downloader::EraClient;
+use std::{
+    collections::HashMap,
+    io::Cursor,
+    path::{Path, PathBuf},
+    str::FromStr,
+    sync::{Arc, Mutex},
+};
+use tempfile::TempDir;
+use tokio::runtime::Runtime;
+
+/// Default nimbus mainnet url
+/// for downloading mainnet `.era1` files
+const MAINNET_URL: &str = "https://mainnet.era1.nimbus.team/";
+
+/// Succint list of mainnet files we want to download
+/// from <https://mainnet.era1.nimbus.team/>
+/// for testing purposes
+const ERA1_FILES_NAMES: [&str; 3] =
+    ["mainnet-00000-5ec1ffb8.era1", "mainnet-00003-d8b8a40b.era1", "mainnet-00151-e322efe1.era1"];
+
+/// Helper for downloading `.era1` files for tests
+/// in a temporary directory
+/// and caching them in memory
+#[derive(Debug)]
+pub struct Era1TestDownloader {
+    temp_dir: TempDir,
+    runtime: Runtime,
```

Comment:
what we could do is use `#[tokio::test(flavor = mulithread)]` for tests 

we could either use a tokio handle here or just make all of the test fn async, or use block_in_place